### DISCUSSION TITLE
Feature to sort JSON object by keys

### DISCUSSION THREAD

**Discussion by leodmgs** [ajaxorg/ace#5054] [Ideas]
Hello,

I was looking for a feature to sort a JSON object by key names alphabetically. In a scenario we have thousands of entries it would be great to have this feature and then easily locate the key we are searching for. I know that I can use the "search" feature, but another case could be to compare two objects. For example, in my case, I have two apps -- each of them have configs stored in the database -- and I want to compare both configs by walking though it.

This is my 2 cents of idea. 😄

**Comment by nightwing** [comment]
You could use the replacer parameter of `JSON.stringify` to sort config before passing it to the editor
```js
var unordered = JSON.parse(text)
var newText = JSON.stringify(unordered, function(key, value) {
    if (value && typeof value == "object" && !Array.isArray(value))
        return Object.fromEntries(Object.entries(value).sort())
    return value;
}, 4)
editor.setValue(newText)
```

**Comment by leodmgs** [comment]
Hey @nightwing,

Sure we can do that way, but I was thinking in keep the configs in the same order as it was stored and just providing an option to view it in an ordered manner -- then kinda "view original" to move it back to the original order.

BTW, thanks for the feedback!

**Reply by nightwing** [reply]
Trying to maintain the original order introduces many tricky edge cases, like where to put added new keys, how to handle json that is not formatted consistently, etc. In your application where you have more specific requirements, you can use another editor.session to display the sorted json, which will be readonly and then when user attempts to edit you can return the original, or you can restore the original order when saving.
